src/System.IO.FileSystem/src/System/IO/PathHelpers.cs

```diff
@@ -81,9 +64,9 @@ internal static string CombineNoChecks(ReadOnlySpan<char> first, ReadOnlySpan<ch
         /// Combines three paths. Does no validation of paths, only concatenates the paths
         /// and places a directory separator between them if needed.
         /// </summary>
-        internal static string CombineNoChecks(string first, ReadOnlySpan<char> second, ReadOnlySpan<char> third)
+        internal static string CombineNoChecks(ReadOnlySpan<char> first, ReadOnlySpan<char> second, ReadOnlySpan<char> third)
         {
-            if (string.IsNullOrEmpty(first))
```

Comment:
If there were any other places you were doing IsNullOrEmpty on these, presumably they woudl continue to compile silently..

Reply:
There's no cast from a span to a string, so I'd expect such usage to fail.

Reply:
You're right.